banking-app/scripts/test_mccf.sh

```diff
@@ -0,0 +1,68 @@
+#!/bin/bash
+set -euo pipefail
+
+declare certificate_dir="./workspace/mccf_certificates"
+
+function usage {
+    echo ""
+    echo "Start a CCF node in mCCF."
+    echo ""
+    echo "usage: ./test_mccf.sh --address <ADDRESS>"
+    echo ""
+    echo "  --address  string      The address of the primary CCF node"
+    echo ""
+    exit 0
+}
+
+function failed {
+    printf "💥 Script failed: %s\n\n" "$1"
+    exit 1
+}
+
+# parse parameters
+
+if [ $# -gt 2 ]; then
+    usage
+    exit 1
+fi
+
+while [ $# -gt 0 ]
+do
+    name="${1/--/}"
+    name="${name/-/_}"
+    case "--$name"  in
+        --address) address="$2"; shift;;
+        --help) usage; exit 0;;
+        --) shift;;
+    esac
+    shift;
+done
+
+# validate parameters
+if [ -z $address ]; then
+    failed "You must supply --address"
+fi
+server="https://${address}"
+
+echo "💤 Getting the Service cert from $server"
+# The node is not up yet and the certificate will not be created until it
+# return 200. We can't pass in the ca_cert hence why we use -k
+while [ "200" != "$(curl $server/node/network -k -s -o /dev/null -w %{http_code})" ]
+do
+    sleep 1
+done
+echo "🎉 Got the Service certificate"
+
+mkdir -p ${certificate_dir}
+certAsString=$(curl $server/node/network -k | jq -r .service_certificate)
```

Comment:
I am not 100% sure if this is the proper way of getting service cert (it might be just my lack of understanding).
Is there any chance of man in the middle attack here?

This is the man page of curl.
```
       -k, --insecure
              (TLS) By default, every SSL connection curl makes is verified to be secure. This option allows curl to proceed and operate even for server connections otherwise considered insecure.

              The server connection is verified by making sure the server's certificate contains the right name and verifies successfully using the cert store.

              See this online resource for further details:
               https://curl.haxx.se/docs/sslcerts.html

              See also --proxy-insecure and --cacert.
```

Reply:
I'll try and find another way to get it

Reply:
The service cert is created to disk by the first node: https://microsoft.github.io/CCF/main/operations/certificates.html#service-certificate

Reply:
In the case of a fresh mCCF service, I think it's fine to get the service identity from /service, because what should take place next is that the consortium goes to audit that service (checks attestations, constitution state, initial membership etc), and then signs off on opening iff it meets expectations. The value of the service certificate is an untrusted input.

Reply:
If and only if, sorry.